src/common/k8s-api/endpoints/cluster-role.api.ts

```diff
@@ -4,39 +4,47 @@
  */
 
 import { isClusterPageContext } from "../../utils/cluster-id-url-parsing";
+import type { DerivedKubeApiOptions } from "../kube-api";
 import { KubeApi } from "../kube-api";
+import type { KubeJsonApiData } from "../kube-json-api";
+import type { KubeObjectMetadata } from "../kube-object";
 import { KubeObject } from "../kube-object";
+import type { AggregationRule } from "./types/aggregation-rule";
+import type { PolicyRule } from "./types/policy-rule";
 
-export interface ClusterRole {
-  rules: {
-    verbs: string[];
-    apiGroups: string[];
-    resources: string[];
-    resourceNames?: string[];
-  }[];
+export interface ClusterRoleData extends KubeJsonApiData<KubeObjectMetadata<"cluster-scoped">, void, void> {
+  rules?: PolicyRule[];
+  aggregationRule?: AggregationRule;
 }
 
-export class ClusterRole extends KubeObject {
+export class ClusterRole extends KubeObject<void, void, "cluster-scoped"> {
```

Comment:
Magical strings should be avoided.

Reply:
Sure sure, strings are better than booleans. Both of these have in common that they lead to non OCP complied code which should be avoided as much as possible. In this case it would require splitting `KubeObject` to manageable pieces, so not welcome in this PR.

Reply:
How was this resolved? Unresolving, just in case.